odoo/addons/base/tests/test_tests_tags.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import unittest
+
+from odoo.tests.common import set_tags, check_tags
+
+
+@set_tags('standard', 'nodatabase')
+class TestSetTags(unittest.TestCase):
+
+    def test_set_tags_empty(self):
+        """Test the set_tags decorator with an empty set of tags"""
+
+        @set_tags()
+        class FakeClass():
+            pass
+
+        fc = FakeClass()
+
+        self.assertTrue(hasattr(fc, 'test_tags'))
+        self.assertEqual(fc.test_tags, set())
+
+    def test_set_tags_single_tag(self):
+        """Test the set_tags decorator with a single tag"""
+
+        @set_tags('slow')
+        class FakeClass():
+            pass
+
+        fc = FakeClass()
+
+        self.assertEqual(fc.test_tags, {'slow', })
+
+    def test_set_tags_multiple_tags(self):
+        """Test the set_tags decorator ith multiple tags"""
+
+        @set_tags('slow', 'nightly')
+        class FakeClass():
+            pass
+
+        fc = FakeClass()
+
+        self.assertEqual(fc.test_tags, {'slow', 'nightly'})
+
+
+@set_tags('standard', 'nodatabase')
+class TestCheckTags(unittest.TestCase):
+
+    def setUp(self):
+        class Test_A():
+            pass
+
+        @set_tags('stock')
+        class Test_B():
+            pass
+
+        @set_tags('stock', 'slow')
+        class Test_C():
+            pass
+
+        @set_tags('standard', 'slow')
+        class Test_D():
+            pass
+
+        self.no_tags_obj = Test_A()
+        self.stock_tag_obj = Test_B()
+        self.multiple_tags_obj = Test_C()
+        self.multiple_tags_standard_obj = Test_D()
```

Comment:
Having a separate setup method seems unnecessary here (it's useful for common setup between multiple methods but there's only a single test method)